ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/assertions/CommandDeduplicationAssertions.scala

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.infrastructure.assertions
+
+import java.time.Duration
+
+import com.daml.api.util.DurationConversion
+import com.daml.ledger.api.testtool.infrastructure.Assertions.{assertDefined, fail}
+import com.daml.ledger.api.testtool.infrastructure.participant.{
+  CompletionResponse,
+  ParticipantTestContext,
+}
+import com.daml.ledger.api.v1.experimental_features.CommandDeduplicationPeriodSupport.{
+  DurationSupport,
+  OffsetSupport,
+}
+import com.daml.ledger.client.binding.Primitive.Party
+import com.daml.platform.testing.WithTimeout
+import com.google.protobuf.duration.{Duration => DurationProto}
+import io.grpc.Status.Code
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+import scala.math.Ordering.Implicits._
+
+object CommandDeduplicationAssertions {
+
+  def assertDeduplicationDuration(
+      requestedDeduplicationDuration: DurationProto,
+      completionResponse: CompletionResponse,
+      submittingParty: Party,
+      ledger: ParticipantTestContext,
+  )(implicit executionContext: ExecutionContext): Future[Unit] = {
+    val requestedDuration = DurationConversion.fromProto(requestedDeduplicationDuration)
+    ledger.features.commandDeduplicationFeatures.getDeduplicationPeriodSupport.durationSupport match {
+      case DurationSupport.DURATION_NATIVE_SUPPORT =>
+        val reportedDurationProto = assertDefined(
+          completionResponse.completion.deduplicationPeriod.deduplicationDuration,
+          "No deduplication duration has been reported",
+        )
+        val reportedDuration = DurationConversion.fromProto(reportedDurationProto)
+        assert(
+          reportedDuration >= requestedDuration,
+          s"The reported deduplication duration $reportedDuration was smaller than the requested deduplication duration $requestedDuration.",
+        )
+        Future.unit
```

Comment:
Based on my tests it works fine. I wanted to avoid a `Future` block and extra scheduling. I will do it, as you're right, it's probably a cleaner approach.